### PR TITLE
chore: unit test cases for check and launch dependency

### PR DESCRIPTION
**Pull request by manikandan-thangavelu-rl**
Issue #, if available:

Description of changes:

Unit test cases for check and launch dependencies
Fix: Fetching ALB stack failure reasons using API

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    });
+
+    it('should throw error obtaining lock fails', async () => {
+      lockService.tryWriteLock.mockImplementationOnce(() => {
+        return undefined;
+      });
+      await expect(step.start()).rejects.toThrow('Could not obtain a lock');
+    });
+
+    it('should call provisionAlb when template output has needsALB set to true', async () => {
```
Comment (jn1119) [inline addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js:142]: I thought albLock also needs to have some value for this test to pass, no?

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+    it('should call alb deployment when alb does not exists', async () => {
+      albService.albDependentWorkspacesCount.mockImplementationOnce(() => {
+        return 1;
+      });
+      albService.checkAlbExists.mockImplementationOnce(() => {
+        return false;
+      });
+      jest.spyOn(step, 'deployStack').mockImplementationOnce(() => {});
+      await step.provisionAlb(requestContext, resolvedVars, 'test-project-id', [], 10);
```
Comment (jn1119) [inline addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js:224]: Same here- not sure. Is ALB_LOCK already set for this to pass?

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+jest.mock('@aws-ee/base-services/lib/lock/lock-service');
+const LockServiceMock = require('@aws-ee/base-services/lib/lock/lock-service');
+
+jest.mock('../../../../environment-type-mgmt-services/lib/environment-type/env-type-config-service.js');
+const EnvTypeConfigServiceMock = require('../../../../environment-type-mgmt-services/lib/environment-type/env-type-config-service.js');
+
+const CheckLaunchDependency = require('../check-launch-dependency/check-launch-dependency');
+
+describe('CheckLaunchDependencyStep', () => {
```
Comment (jn1119) [inline addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js:30]: Add test for `parseS3DetailsfromUrl`?

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js [hunk 4]
```diff
@@ -367,7 +385,8 @@ class TerminateLaunchDependency extends StepBase {
     if (_.includes(failureStatuses, stackInfo.StackStatus)) {
       // If termination failed then throw error, any unhandled workflow errors
       // are handled in "onFail" method
```
Comment (jn1119) [inline addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js:387]: add a test for `onFail`, `onPass` and the timeout method for both terminate  and launch steps?

**Review by jn1119 — APPROVED** [review]
For addressing PR comments, I think we can use another PR. I am going to approve this change for now.

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js (hunk 1 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js:142]: We have a mock function to set ALB_LOCK.

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js (hunk 2 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js:224]: Yes, We have a mock function to set ALB_LOCK.

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js (hunk 3 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/check-launch-dependency.test.js:30]: We will add the test case and raise another PR

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js (hunk 4 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js:387]: We will add the test case and raise another PR

**Pull request merged**